coordinator/coordinator.go

```diff
@@ -76,15 +90,23 @@ func (c *Coordinator) Run() error {
 		status := c.cluster.Status()
 		mlog.Info("coordinator: cluster status:", mlog.Int("active_users", status.ActiveUsers), mlog.Int64("errors", status.NumErrors))
 
-		// TODO: supportedUsers should be estimated in a more clever way in the future.
-		// For now we say that the supported number of users is the number of active users that ran
-		// for the defined timespan without causing any performance degradation alert.
-		if !lastAlertTime.IsZero() && !perfStatus.Alert && hasPassed(lastAlertTime, restTime) && hasPassed(lastActionTime, restTime) {
-			supportedUsers = status.ActiveUsers
+		if !lastAlertTime.IsZero() {
+			samples = append(samples, point{
+				x: time.Now(),
+				y: status.ActiveUsers,
+			})
+			latest := getLatestSamples(samples, samplesTimeRange)
+			if len(latest) > 0 && len(latest) < len(samples) && math.Abs(slope(latest)) < stopThreshold {
+				mlog.Info("coordinator done!")
+				mlog.Info(fmt.Sprintf("estimated number of supported users is %f", math.Round(avg(latest))))
+				return nil
+			}
+			// We truncate to remove older samples which are not needed anymore.
+			if len(samples) >= 2*len(latest) {
```

Comment:
Ok, I want to confirm something here. Because I think it's not doing what we want it to do here.

If we re-slice from the end of a slice, does it actually reuse the free space? I don't think so right? In that case, it becomes the same problem as before. We might need to re-allocate a fresh-slice to free up the older one.

Reply:
That's a valid concern but I am thinking the call to `append` will take care of that in the long term. At some point a new array will be created and the old one will get garbage collected.
If that seems too obscure as a reasoning I'll just make a brand new copy :)

Reply:
That also works, but I think the best and the neatest option might be to just overwrite the front portion of the slice with the new values, and then just re-slice till that index. That solves everything.